eo-parser/src/main/java/org/eolang/parser/StHash.java

```diff
@@ -0,0 +1,119 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.parser;
+
+import com.jcabi.xml.XML;
+import com.jcabi.xml.XMLDocument;
+import com.yegor256.xsline.StEnvelope;
+import com.yegor256.xsline.StLambda;
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import org.xembly.Directives;
+import org.xembly.Xembler;
+
+/**
+ * Add attribute 'hash' in node 'program' in XML.
+ * Returns already XML with hash.
+ *
+ * @since 0.35.0
+ *
+ * @todo #2764:30min Replace the hash code entry
+ *  from the XML file node attribute to the file attribute.
+ *  This is necessary to increase the speed of searching for a cached file.
+ *  The hash code entry should be deleted from EoSyntax.java and PhiSyntax.java.
+ *  Hash code should save to the file attribute in ParseMojo.java and UnphiMojo.
+ *  Delete tests containsHash() from EoSyntaxTest.java and PhiSyntaxTest.java.
+ * @todo #2764:30min Is it possible set the hash function directly
+ *  in XeEoListener during the parsing? Now hash code is generated in class StHash.java.
+ *  Maybe it will be better generating hash code in class XeEoListener.
+ */
+public final class StHash extends StEnvelope {
+
+    /**
+     * Returns XML with attribute 'hash'.
+     */
+    public StHash() {
+        super(
+            new StLambda((
+                position,
+                xml) ->
+                new XMLDocument(
+                    new Xembler(
+                        new Directives().xpath("//program").attr("hash", new Hash(xml).compute())
+                    ).apply(xml.node())
+                )
+            )
+        );
+    }
+
+    /**
+     * Hash is generated by MD5 using node '/program/objects'.
+     * Returns program hash.
+     *
+     * @since 0.35.0
+     */
+    public static final class Hash {
+
+        /**
+         * XML for which the hash code will be made.
+         */
+        private final XML xml;
+
+        /**
+         * The constructor.
+         *
+         * @param xml XML for which the hash code will be made.
+         */
+        public Hash(final XML xml) {
+            this.xml = xml;
+        }
+
+        /**
+         * Return program hash using node "/program/objects".
+         *
+         * @return String hash of this XML.
+         * @throws NoSuchAlgorithmException If fails.
+         */
+        public String compute() throws NoSuchAlgorithmException {
+            final BigInteger number = new BigInteger(
+                1,
+                MessageDigest
+                    .getInstance("MD5")
+                    .digest(
+                        this.xml.nodes(
+                            "/program/objects"
+                        )
+                            .toString().getBytes()
```

Comment:
@Yanich96 Here you apply `toString` method to a `List` instance, which is "ok" in this case (since the list contains strings.) However, why don't you use just `xml.toString` ? It looks simpler and doesn't require additional computations like applying paths

Reply:
@volodya-lombrozo I use '/program/objects' for creating the hash code. If 'xml.toString' is used, hash code will be created from all xml.

Reply:
@Yanich96 Why?